documents/coredocument.go

```diff
@@ -528,3 +537,22 @@ func (cd *CoreDocument) Signatures() (signatures []coredocumentpb.Signature) {
 
 	return signatures
 }
+
+func populateVersions(cd *coredocumentpb.CoreDocument, prevCD *coredocumentpb.CoreDocument) (err error) {
+	if prevCD != nil {
+		cd.PreviousVersion = prevCD.CurrentVersion
+		cd.CurrentVersion = prevCD.NextVersion
+		cd.CurrentPreimage = prevCD.NextPreimage
```

Comment:
okay so we are storing the current preImage and the nextPreimage inside the CD.

Need to update that in the protocol paper

Reply:
we have to,